packages/internal/src/generate/graphqlSchema.ts

```diff
@@ -1,59 +1,102 @@
 
 import { rootSchema } from '@redwoodjs/graphql-server'
 
 import { getPaths } from '../paths'
 
 export const generateGraphQLSchema = async () => {
-  const rwjsPaths = getPaths()
-  type GenerateResponse = { filename: string; contents: string }[]
-  try {
-    const f: GenerateResponse = await generate(
-      {
-        cwd: rwjsPaths.api.src,
-        schema: [
-          print(rootSchema.schema), // generator cli needs a string or a filePath
-          'graphql/**/*.sdl.{js,ts}',
-          'directives/**/*.{js,ts}',
-        ],
-        config: {
-          scalars: {
-            BigInt: 'number',
-            DateTime: 'string',
-            Date: 'string',
-            JSON: 'Record<string, unknown>',
-            JSONObject: 'Record<string, unknown>',
-            Time: 'string',
-          },
-        },
-        generates: {
-          // Note the plugin passed, we're not generating types here.
-          // its subtley different to generateTypeDefGraphQL
-          [rwjsPaths.generated.schema]: {
-            plugins: ['schema-ast'],
-          },
-        },
-        silent: false,
-        errorsOnly: true,
+  const schemaPointerMap = {
+    [print(rootSchema.schema)]: {},
+    'graphql/**/*.sdl.{js,ts}': {},
+    'directives/**/*.{js,ts}': {},
+  }
+
+  const config = {
+    scalars: {
+      BigInt: 'number',
+      DateTime: 'string',
+      Date: 'string',
+      JSON: 'Record<string, unknown>',
+      JSONObject: 'Record<string, unknown>',
+      Time: 'string',
+    },
+  }
+
+  const loadSchemaConfig: LoadSchemaOptions = {
+    assumeValidSDL: true,
+    sort: true,
+    convertExtensions: true,
+    includeSources: true,
+    cwd: getPaths().api.src,
+    schema: Object.keys(schemaPointerMap),
+    config,
+    generates: {
+      [getPaths().generated.schema]: {
+        plugins: ['schema-ast'],
       },
-      true
-    )
-    return f[0].filename
-  } catch (e: any) {
-    // `generate` outputs errors which are helpful.
-    // This tries to clean up the output of those errors.
-    console.error(chalk.red('Error parsing SDLs or Schema'))
-    if (e.errors?.length) {
-      for (const error of e.errors ?? []) {
-        console.error(error.details)
+    },
+    silent: false,
+    errorsOnly: false,
+    pluginContext: {},
+    loaders: [new CodeFileLoader()],
+  }
+
+  let loadedSchema
+
+  try {
+    loadedSchema = await loadSchema(schemaPointerMap, loadSchemaConfig)
+  } catch (e) {
+    if (e instanceof Error) {
+      const match = e.message.match(/Unknown type: "(\w+)"/)
+      const name = match?.[1]
+      const schemaPrisma = fs.readFileSync(getPaths().api.dbSchema)
+
+      console.error('Schema loading failed.', e.message)
+
+      if (name && schemaPrisma.includes(`model ${name}`)) {
+        // Not all SDLs need to be backed by a DB model, but if they are we can
+        // provide a more helpful error message
+
+        console.error('')
+        console.error(
```

Comment:
@dthyresson Can you please help me out with what kind of messaging we want here?

Reply:
Hi @Tobbe ideally we'd share some language and link to this doc: https://github.com/redwoodjs/redwood/pull/4849/files

But waiting on @jtoar or others to review and publish.

> If you see `Error: Unknown type: ...`, don't panic!
> It's a known issue with GraphQL type generation.
> It happens when you generate the SDL of a Prisma model that has relations **before the SDL for the related model** exists.
> Please see [How to Fix Type Generation Error with Relations​](./schema-relations#errors-when-generating-sdl-or-scaffolds-for-relations) for help.

Reply:
@Tobbe I just merged https://github.com/redwoodjs/redwood/pull/4849

Reply:
@jtoar Thanks. I added a link to the terminal output